h/h_api/model/json_api.py

```diff
@@ -44,8 +44,20 @@ class JSONAPIData(Model):
 
     @classmethod
     def create(
-        cls, data_type, _id=None, attributes=None, meta=None, relationships=None
+        cls,
+        data_type,
+        _id=None,
+        attributes=None,
+        meta=None,
+        relationships=None,
+        id_reference=None,
```

Comment:
Moved `id_reference` here as this class has the property for reading it. Previously the upsert was responsible for setting it, which is a bit odd. All commands can theoretically have this, although only users and groups do right now.